### ISSUE TITLE
[24] test failures in ep434Y-unit-cen64-gtk3-java24

### ISSUE THREAD

**Issue by stephan-herrmann** [eclipse-pde/eclipse.pde#1462]
Build job https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24 runs tests with
* BEAT_JAVA24 branches from JDT
* JDK 24 EA

Since build https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10 we are down to a manageable number for failures. These are in PDE:
* [ClasspathResolutionTest.testImportSystemPackageDoesntAddExtraBundleJava8](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.core.tests.internal.classpath/ClasspathResolutionTest/testImportSystemPackageDoesntAddExtraBundleJava8/)
* [ClasspathResolutionTest.testImportSystemPackageDoesntAddExtraBundleJava8_osgiEERequirement](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.core.tests.internal.classpath/ClasspathResolutionTest/testImportSystemPackageDoesntAddExtraBundleJava8_osgiEERequirement/)
* [BundleErrorReporterTest.testErrorOnUnresolvedJrePackage](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.core.tests.internal.core.builders/BundleErrorReporterTest/testErrorOnUnresolvedJrePackage/)
* [TargetEnvironmentTestCase.testProjectWithJVMImports](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.ui.tests.target/TargetEnvironmentTestCase/testProjectWithJVMImports/)
* [TestPDETemplates.validateProduct](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/(root)/[21_%20Eclipse%20RCP%20application]/validateProduct_21__Eclipse_RCP_application_/)

It would be good to know what kind of failures we have here:
* do any failures indicate a problem in jdt, perhaps jdt.launching?
* can failures be resolved in pde master?
* do failures indicate that also PDE would need a BETA_JAVA24 branch for tests on Java 24 to succeed?

For investigation you might want to pull JDT beta artifacts from https://download.eclipse.org/eclipse/updates/4.34-Y-builds/

**Comment by stephan-herrmann** [comment]
Meanwhile the build to observe is https://ci.eclipse.org/releng/job/YPBuilds/job/ep435Y-unit-linux-x86_64-java24/

And here's a recent test report for pde (13 failures): https://download.eclipse.org/eclipse/downloads/drops4/Y20250106-1000/testresults/html/org.eclipse.pde.ui.tests_ep435Y-unit-linux-x86_64-java24_linux.gtk.x86_64_24.html 

JDT Beta builds are now in https://download.eclipse.org/eclipse/updates/4.35-Y-builds/

**Comment by jukzi** [comment]
BundleErrorReporterTest.testErrorOnUnresolvedJrePackage also failed in I-Build https://download.eclipse.org/eclipse/downloads/drops4/I20250107-1800/testresults/html/org.eclipse.pde.ui.tests_ep435I-unit-macosx-x86_64-java21_macosx.cocoa.x86_64_21.html

```
java.lang.AssertionError:
Expecting empty but was: [Marker [on: /org.eclipse.pde.core.tests.internal.core.builders.BundleErrorReporterTest/META-INF/MANIFEST.MF, id: 570, type: org.eclipse.pde.core.problem, attributes: [categoryId: fatal, compilerKey: compilers.p.unresolved-import, lineNumber: 6, message: Neither an available bundle nor the associated JRE export package 'java.lang.module', packageName: java.lang.module, problemId: 4103, severity: 2], created: 1/7/25, 10:10 PM]]
at org.eclipse.pde.core.tests.internal.core.builders.BundleErrorReporterTest.testErrorOnUnresolvedJrePackage(BundleErrorReporterTest.java:79)
```

**Comment by stephan-herrmann** [comment]
> Since build https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10 we are down to a manageable number for failures. These are in PDE:
> * [ClasspathResolutionTest.testImportSystemPackageDoesntAddExtraBundleJava8](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.core.tests.internal.classpath/ClasspathResolutionTest/testImportSystemPackageDoesntAddExtraBundleJava8/)
> * [ClasspathResolutionTest.testImportSystemPackageDoesntAddExtraBundleJava8_osgiEERequirement](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.core.tests.internal.classpath/ClasspathResolutionTest/testImportSystemPackageDoesntAddExtraBundleJava8_osgiEERequirement/)
> * [BundleErrorReporterTest.testErrorOnUnresolvedJrePackage](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.core.tests.internal.core.builders/BundleErrorReporterTest/testErrorOnUnresolvedJrePackage/)
> * [TargetEnvironmentTestCase.testProjectWithJVMImports](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/org.eclipse.pde.ui.tests.target/TargetEnvironmentTestCase/testProjectWithJVMImports/)
> * [TestPDETemplates.validateProduct](https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/10/testReport/(root)/%5B21_%20Eclipse%20RCP%20application%5D/validateProduct_21__Eclipse_RCP_application_/)

Of these, the [latest build](https://ci.eclipse.org/releng/job/YPBuilds/job/ep435Y-unit-linux-x86_64-java24/34/testReport/) still shows four failures. Only the last one, TestPDETemplates.validateProduct, is no longer listed as failing.

Did anyone have a look if any of these indicate that PDE needs some changes of its own to work with Java 24?

**Comment by HannesWell** [comment]
> Did anyone have a look if any of these indicate that PDE needs some changes of its own to work with Java 24?

Not yet in detail, but after a quick glance the first two look related and for the fourth I wonder how this even passes in current I-builds as it uses a project with Java 1.7 BREE.

I'll try to have a deeper look in the next days.

**Comment by stephan-herrmann** [comment]
@HannesWell in [yesterdays tests](https://ci.eclipse.org/releng/job/YPBuilds/job/ep435Y-unit-linux-x86_64-java24/45/testReport/) PDE is clean!
I guess this may be caused by https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/616 - does this make sense wrt the nature of previous failures?

Do you agree to closing this issue then?

**Comment by stephan-herrmann** [comment]
Subsequent builds speak the same language: problem resolved.

**Comment by HannesWell** [comment]
> I guess this may be caused by [eclipse-jdt/eclipse.jdt.debug#616](https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/616) - does this make sense wrt the nature of previous failures?

Yes this sounds related. Thanks for the fix and sorry for the late reply.
I'm glad that PDE is now on track for Java-24 as well (without having to do anything).
